### PR TITLE
Add argocd chart

### PR DESCRIPTION
**Pull request by mduarte**

### CODE REVIEW

File: argocd/templates/application-crd.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+kind: CustomResourceDefinition
+metadata:
+  name: applications.argoproj.io
+spec:
+  group: argoproj.io
+  names:
+    kind: Application
+    plural: applications
+    shortNames:
+    - app
```
Comment (conorfennell) [inline argocd/templates/application-crd.yaml]: add apps https://github.com/argoproj/argo-cd/blob/d6c88cd77a39737b84cff02025aaf81bb24f6f57/manifests/crds/application-crd.yaml#L14 :)

File: argocd/templates/argocd-application-controller-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      labels:
+        app: argocd-application-controller
+    spec:
+      containers:
+      - command:
+        - argocd-application-controller
+        - --status-processors
+        - "20"
+        - --operation-processors
+        - "10"
```
Comment (conorfennell) [inline argocd/templates/argocd-application-controller-deployment.yaml]: is this the number of threads and an arbitrary pick?

File: argocd/templates/argocd-repo-server-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  replicas: {{ .Values.repoServer.replicas }}
+  selector:
+    matchLabels:
+      app: argocd-repo-server
+  template:
+    metadata:
+      labels:
+        app: argocd-repo-server
+    spec:
+      automountServiceAccountToken: false
```
Comment (conorfennell) [inline argocd/templates/argocd-repo-server-deployment.yaml]: ?

File: argocd/templates/argocd-application-controller-deployment.yaml (hunk 2 above)
Comment (mduarte) [reply argocd/templates/argocd-application-controller-deployment.yaml]: This has been copied over from their manifests assuming that there's been some reasoning behind the values that they are putting there.

File: argocd/templates/argocd-repo-server-deployment.yaml (hunk 3 above)
Comment (mduarte) [reply argocd/templates/argocd-repo-server-deployment.yaml]: This has also been copied over from their manifests without digging deep into the reasoning behind it. However, a quick look into  https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/ gives us:
> In version 1.6+, you can opt out of automounting API credentials for a service account by setting `automountServiceAccountToken: false` on the service account

Comment (conorfennell) [reply argocd/templates/argocd-repo-server-deployment.yaml]: Thought there might be something behind it, its like if your pod does not interact with the kube api you should always set this to false... all good.

**Pull request merged**